addon/index.js

```diff
@@ -50,7 +50,24 @@ export default function deepSet(obj, key, value) {
       allKeys[i + 1] && isObject(currentValue)  ? currentValue :
       allKeys[i + 1] && !isObject(currentValue) ? {} :
       value;
-    return set(acc, currentKey, valueToSet);
+    if (valueToSet === undefined) {
+      // ember's set method does not handle undefined values correctly in older versions
+      // https://github.com/emberjs/ember.js/issues/14270
+      if (
+        acc.hasOwnProperty(currentKey) ||
```

Comment:
@jelhan Want to make sure of your intention - should this be `&&` or `||`?

Reply:
It's by intention. Maybe it would be easier to read the other way around: `setUnknownProperty` should only be used if it exists and if the property is not declared on object.

Reply:
I think this pseudo polyfill for `set` is a-ok.  Nice work!  Just wanted to make sure.